### PR TITLE
Push Docker images to CERN registry

### LINKED ISSUE #51: Base Docker image
- [x] Look into [conversation on Discord](https://discord.com/channels/692989811736182844/819158359147675648/1006842960735903764) with Werner about using the RedHat UBI image
- [x] See existing [docker-invenio](https://github.com/inveniosoftware/docker-invenio)
- [POSTPONED] How to include XRootD distribution in the image
   - Out of scope (because of complexity); we'll use local files instead (common R/W volume on OpenShift)
   - Look into S3-based file API?

**Decision:** use `AlmaLinux` as base image.

### Findings

Today (20 Oct 2022), apps end of life:
- Python
  - 3.10 in ~4 years
  - 3.9 in ~3 years
  - 3.8 in 2 years
  - 3.7 in 8 months
- NodeJs
  - v18: upcoming
  - v16: ended, only security updates for 10 months
- Almalinux
  - v9: in 4.5 years

### Tasks

- [x] Review exiting Docker image with AlmaLinux and Python 3.9
  - for example, why did we pin setuptools?
  - run as normal user, not root? WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager
  - check if any vulnerabilities: https://registry.cern.ch/harbor/projects/1825/
- [x] Decide the naming/tagging convention, for example `inveniosoftware/almalinux9:python3.9-node18` vs `inveniosoftware/almalinux9-python3.9-node18` vs .... Ask the community.
  - Decided to go for incremental versions instead: `inveniosoftware/almalinux:v1.0.0`
- [x] Can we host such images in [registry.cern.ch](https://registry.cern.ch), to avoid DockerHub limitations? Is there any other alternative?
- [x] Setup automatic builds

### PR DESCRIPTION
**Pull request by ntarocco**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Release
+
+on:
+  push:
```
Comment (avivace) [inline .github/workflows/release.yml:4]: Maybe we could also run (some of) the actions here on `pull_request`? (E.g. just to build and test the images, without publishing them)

File: almalinux/Dockerfile [hunk 2]
```diff
@@ -7,7 +7,8 @@
 # under the terms of the MIT License; see LICENSE file for more details.
```
Comment (avivace) [inline almalinux/Dockerfile:7]: Why this has been moved? Do we plan to change the current structure of the repository? How/where would we put e.g. an AlmaLinux image targeting python3.8?

**Review by avivace — APPROVED** [review]
LGTM! I just left a couple of minor remarks.

File: .github/workflows/release.yml (hunk 1 above)
Comment (ntarocco) [reply .github/workflows/release.yml:4]: Yes, good point. We will do that as soon as we fix the test for the Almalinux image.

File: almalinux/Dockerfile (hunk 2 above)
Comment (ntarocco) [reply almalinux/Dockerfile:7]: Well, I did this after the chat we had: the idea is to provide one image, versioned, that works. For example `inveniosoftware/almalinux:v1.0.0`: such image can contain, for example, AlmaLinux 9, Python 3.9, NodeJS 18.
Upgrading it to Python 3.10 would then mean that the tag will become `v2.0.0`.
@mfenner is my interpretation of our chat correct here?

It looks like that we don't want to create, maintain and publish a matrix of images.

Comment (mfenner) [reply almalinux/Dockerfile:7]: Yes. We didn‘t discuss the version numbering, e.g. whether Python 3.10 would mean v2.0.0 or v1.1.0, but that is an implementation detail. Would follow the SemVer best practices, so minor image version bumps for version updates of dependencies, and major version bumps for breaking changes.

Comment (ntarocco) [reply almalinux/Dockerfile:7]: Yes of course, this is what I meant, mine was just an example. @avivace will this work for you?

**Review by mfenner — APPROVED** [review]
LGTM. Minor points: 

* trigger GitHub Action to run tests on pull-requests as suggested by @avivace
*  think about the matrix for running tests (e.g. do we need Python 3.6, what do with CentOS 8)
* I would enable multi-platform builds (linux/amd64, linux/arm64), as I expect there will be small but growing use of M1/M2 Macs of these images

**Comment by ntarocco** [comment]
I will merge this, @avivace will continue adding the missing parts:
* trigger GitHub Action to run tests on pull-requests
* fix/improve Almalinux image
* fix tests for Almalinux

For the CentOS 8, after a few discussions, we can:
* in this repo, remove or "deprecate" them with a clear msg
* on Docker Hub, we will add a clear message in the README to explain what happened

**Pull request merged**